api/v1alpha1/runtime.proto

```diff
@@ -127,6 +127,16 @@ enum ControllerInputKind {
     //
     // Controller can add finalizers on strong inputs.
     STRONG = 1;
+    // DestroyReady inputs provide notifications to the controllers when some resource
+    // is ready to be destroyed (all finalizers are removed).
+
+    // Usually outputs are declared as DestroyReady inputs to trigger reconcile loop
+    // when resource is ready to be destroy.
+    //
+    // Controller will be notified when the resource enters "teardown" phase and has no
+    // finalizers attached.
+    // Resources are filtered to be owned by the controller.
+    DESTROY_READY = 2;
```

Comment:
This feels a little out of place here, but I am not comfortable enough with the code yet to suggest anything different.

Reply:
I was going back and forth on the design here, as it certainly feels like an output property, but outputs don't have namespaces, so I got stuck as a watch implies having a namespace.

Reply:
probably we can come up with some other concept here to express it better

Reply:
Where did we land with this?

Reply:
I don't have any better idea than what we have here, it's more about the API, less about the actual implementation.